libc/cmake/modules/LLVMLibCTestRules.cmake

```diff
@@ -46,13 +46,13 @@ function(_get_common_test_compile_options output_var c_test flags)
       list(APPEND compile_options "-Wno-gnu-imaginary-constant")
     endif()
     list(APPEND compile_options "-Wno-pedantic")
-    # if(NOT CMAKE_COMPILER_IS_GNUCXX)
-    #   list(APPEND compile_options "-Wnewline-eof")
-    #   list(APPEND compile_options "-Wnonportable-system-include-path")
-    #   list(APPEND compile_options "-Wstrict-prototypes")
-    #   list(APPEND compile_options "-Wthread-safety")
-    #   list(APPEND compile_options "-Wglobal-constructors")
-    # endif()
+    if(NOT CMAKE_COMPILER_IS_GNUCXX)
```

Comment:
If the below command line flags are clang specific, mind changing `NOT CMAKE_COMPILER_IS_GNUCXX` to `CMAKE_CXX_COMPILER_ID STREQUAL "Clang"`?

https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html

!gcc != clang

even if today we only support those two.

---
`-Wstrict-prototypes` is supported by both compilers, so it should not be in a clang specific list. Rather, it should be enabled for both compilers. Please move it up to the block on L36-L37.

Reply:
Fixed in be9ec1ac8875d5d4e32c5be5095a60da5fd02e12

Reply:
> -Wstrict-prototypes is supported by both compilers, so it should not be in a clang specific list. Rather, it should be enabled for both compilers. Please move it up to the block on L36-L37.

Ah, shoot! Something I missed was:

```
cc1plus: error: command-line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
```
So while GCC _does_ support `-Wstrict-prototypes`, it only does so for C mode (not C++).  So sorry for the goose chase, but my previous feedback here was wrong and bad advice. Do you mind moving `-Wstrict-prototypes` back to just being set for clang?

Reply:
I observe
```
cc1: error: command-line option ‘-Wextra-semi’ is valid for C++/ObjC++ but not for C [-Werror]
```
when trying to build llvm-project/libc/test/include/stdbit_test.c. That's a pre-existing issue that predates your PR, but it's the same issue as I've outlined above regarding `-Wstrict-prototypes`. While you're "in the neighborhood," do you mind please also moving `-Wextra-semi` back to being Clang-only together with `-Wstrict-prototypes`? For the same reasons.  Ideally, would have C specific compiler flags and C++ specific compiler flags, but I'm not about to "shave that yak."